src/main/java/org/springframework/data/redis/connection/RedisStreamCommands.java

```diff
@@ -183,16 +424,26 @@ public static XAddOptions maxlen(long maxlen) {
 		 * @since 2.7
 		 */
 		public XAddOptions minId(RecordId minId) {
-			return new XAddOptions(maxlen, nomkstream, approximateTrimming, minId);
+			return new XAddOptions(nomkstream, TrimOptions.minId(minId));
 		}
 
 		/**
 		 * Apply efficient trimming for capped streams using the {@code ~} flag.
 		 *
 		 * @return new instance of {@link XAddOptions}.
+		 * @deprecated since 4.0: callers must specify a concrete trim strategy (MAXLEN or MINID)
+		 * via {@link TrimOptions}; do not use this method to only toggle approximate/exact.
+		 * Prefer {@code XAddOptions.trim(TrimOptions.maxLen(n).approximate())} or
+		 * {@code XAddOptions.trim(TrimOptions.minId(id).exact())}.
 		 */
+		@Deprecated(since = "4.0", forRemoval = false)
```

Comment:
I suggest avoiding `forRemoval = true`. It's a pretty harsh signal. Just the deprecation is a lighter form of signalling improper API usage, and that there is a better variant.